hosted/aks/helper/helper_cluster.go

```diff
@@ -108,75 +206,173 @@ func GetK8sVersionVariantAKS(minorVersion string, client *rancher.Client, cloudC
 }
 
 // AddNodePool adds a nodepool to the list
-func AddNodePool(cluster *management.Cluster, increaseBy int, client *rancher.Client) (*management.Cluster, error) {
+func AddNodePool(cluster *management.Cluster, increaseBy int, client *rancher.Client, wait, check bool) (*management.Cluster, error) {
+	currentNodePoolNumber := len(cluster.AKSConfig.NodePools)
+
 	upgradedCluster := new(management.Cluster)
 	upgradedCluster.Name = cluster.Name
 	upgradedCluster.AKSConfig = cluster.AKSConfig
-	nodeConfig := AksHostNodeConfig()
 
+	updateNodePoolsList := cluster.AKSConfig.NodePools
 	for i := 1; i <= increaseBy; i++ {
-		for _, np := range nodeConfig {
+		for _, np := range cluster.AKSConfig.NodePools {
 			newNodepool := management.AKSNodePool{
 				Count:             pointer.Int64(1),
 				VMSize:            np.VMSize,
 				Mode:              np.Mode,
 				EnableAutoScaling: np.EnableAutoScaling,
 				Name:              pointer.String(namegen.RandStringLower(5)),
 			}
-			upgradedCluster.AKSConfig.NodePools = append(upgradedCluster.AKSConfig.NodePools, newNodepool)
+			updateNodePoolsList = append(updateNodePoolsList, newNodepool)
 		}
 	}
-	cluster, err := client.Management.Cluster.Update(cluster, &upgradedCluster)
+	upgradedCluster.AKSConfig.NodePools = updateNodePoolsList
+
+	var err error
+	cluster, err = client.Management.Cluster.Update(cluster, &upgradedCluster)
 	if err != nil {
 		return nil, err
 	}
+
+	if check {
+		// Check if the desired config is set correctly
+		Expect(len(cluster.AKSConfig.NodePools)).Should(BeNumerically("==", currentNodePoolNumber+increaseBy))
+		for i, np := range cluster.AKSConfig.NodePools {
+			Expect(np.Name).To(Equal(updateNodePoolsList[i].Name))
+		}
+	}
+
+	if wait {
+		cluster, err = helpers.WaitClusterToBeUpgraded(client, cluster.ID)
+		if err != nil {
```

Comment:
```suggestion
Expect(err).To(BeNil())
```

Reply:
I intended to use gomega checks for `check` block only. If you don't think it would be a major issue, I would like to keep it as is.

Reply:
I'll prefer to use gomega for uniformity